devito/types/equation.py

```diff
@@ -157,6 +157,211 @@ def __str__(self):
     __repr__ = __str__
 
 
+class Le(sympy.Le):
+    """
+    A less-than ("<=") relation between two objects, the left-hand side and the
+    right-hand side. Can be used to build conditionals but not directly to
+    construct an Operator.
+
+    The left-hand side may be a Function or a SparseFunction. The right-hand
+    side may be any arbitrary expressions with numbers, Dimensions, Constants,
+    Functions and SparseFunctions as operands.
+
+    Parameters
+    ----------
+    lhs : Function or SparseFunction
+        The left-hand side.
+    rhs : expr-like, optional
+        The right-hand side. Defaults to 0.
+    subdomain : SubDomain, optional
+        To restrict the evalaution of the relation to a particular sub-region in the
+        computational domain.
+
+    Examples
+    --------
+    Le may be used to express a relation. Below, a summation along
+    the user-defined Dimension ``i``.
+
+    >>> class InnerDomain(SubDomain):
```

Comment:
I think this example is insanely more complicated than it strictly needs to be

Reply:
SImplified now.